plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -266,8 +275,41 @@ private Client client() {
     @FunctionalInterface
     interface ProcessorFactoryProvider {
 
-        Processor.Factory get(Environment environment);
+        Processor.Factory get(Environment environment, TemplateService templateService);
+
+    }
+
+    static class InternalTemplateService implements TemplateService {
+
+        public static final ScriptContext.Plugin INSTANCE = new ScriptContext.Plugin("elasticsearch-ingest", "ingest");
+
+        private final ScriptService scriptService;
+
+        InternalTemplateService(ScriptService scriptService) {
+            this.scriptService = scriptService;
+        }
 
+        @Override
+        public Template innerCompile(String template) {
+            Script script = new Script(template, ScriptService.ScriptType.INLINE, "mustache", Collections.emptyMap());
```

Comment:
> so we depend on es core which provides us with mustache

No, core doesn't provide it. It merely provides the interfaces script engines can implement. 

> but lang-mustache will be there all the time?

Yes, it is a plugin that is installed by default. So when the ingest plugin starts up we can use it by specifying that we want to use `mustache` language.